controller/api/public/stat_summary.go

```diff
@@ -88,13 +88,23 @@ func (s *grpcServer) StatSummary(ctx context.Context, req *pb.StatSummaryRequest
 		statReq := &pb.StatSummaryRequest{
 			Selector: &pb.ResourceSelection{
 				Resource: &pb.Resource{
-					Type:      resource,
 					Namespace: req.Selector.Resource.Namespace,
+					Type:      resource,
+					Name:      req.Selector.Resource.Name,
 				},
 				LabelSelector: req.Selector.LabelSelector,
 			},
 			TimeWindow: req.TimeWindow,
 		}
+		if req.GetFromResource() != nil {
```

Comment:
Wow! Nice catch!

Another way to implement this would be to copy the entire request struct, so if we ever add more elements to it, we'll be sure to pass it along here:
```go
statReq := *req
statReq.Selector = &pb.ResourceSelection{
	Resource: &pb.Resource{
		Namespace: req.Selector.Resource.Namespace,
		Type:      resource,
		Name:      req.Selector.Resource.Name,
	},
}

go func(sReq pb.StatSummaryRequest) {
	resultChan <- s.resourceQuery(ctx, &sReq)
}(statReq)
```

Reply:
the Clone wars